crates/ruff_linter/resources/test/fixtures/flake8_comprehensions/C417.py

```diff
@@ -47,3 +47,13 @@ def func(arg1: int, arg2: int = 4):
 map(lambda x: x, y if y else z)
 map(lambda x: x, (y if y else z))
 map(lambda x: x, (x, y, z))
+
+# See https://github.com/astral-sh/ruff/issues/14808
+# The following should be Ok since 
+# named expressions are a syntax error inside comprehensions
+a = [1, 2, 3]
+b = map(lambda x: x, c := a)
```

Comment:
Can you add a tested for a nested expression in the target? Like `map(lambda x: x, [c := a])`

Reply:
whoops, I had meant to do something like that in the second example but nested the wrong thing. fixed now, thanks for catching that!